### DISCUSSION TITLE
feature generator: should Boolean and Numerical be stored in the same cache?

### DISCUSSION THREAD

**Discussion by drexlerd** [rleap-project/dlplan#31] [General]
Hi,

I am thinking about merging the caches of `Boolean` and `Numerical` to further reduce the feature pool. When interpreting True=1 and False=0 then every `Boolean` is also a `Numerical`. Beside generalization by luck there is no reason to possibly have both versions in the feature pool at any given time. This won't have a huge impact since the number of `Numerical` is typically quite a bit larger. However, if a `Boolean` is sufficient this modification ensures that our algorithms will pick it.

Any opinions on that?

**Comment by jendrikseipp** [comment]
Merging the caches sounds like a reasonable idea.

**Reply by drexlerd** [reply]
The latest version now creates such a smaller feature pool and it can make sense to rerun previous experiments.

**Reply by drexlerd** [reply]
Some numbers of how this effects the size of the feature pool:

- single Blocksworld instance with 3 blocks and complexity 8:
old feature pool: n_count (2289), b_empty (844), total (3133)
new feature pool: n_count (2177), b_empty (790), total (2967)
roughly 5 percent smaller.

- single 2x2 Delivery instance with 1 package and complexity 12:
old feature pool: n_count (6382), n_concept_distance (4190), b_empty (457), total (11029)
new feature pool: n_count (6276), n_concept_distance (4063), b_empty (358), total (10697)
roughly 3 percent smaller.
